### ISSUE TITLE
Strange behavior picking up large stacks on dedicated server (Fabric 1.20.1)

### ISSUE THREAD

**Issue by EnigmaticWraith** [Tfarcenim/Dank-Storage#276]
When picking up a large stack of items on a dedicated server they are invisible on the cursor. The items can still be placed back in the inventory, although the behavior is glitchy, clicking and dragging only sometimes works (and only after placing one stack) and the number shown is always incorrect. 

Additionally, if you Shift+Throw the stack out of the dank directly, it becomes an invisible item on the ground, which can still be picked up as long as you have a dank that can hold it set to pickup mode.

This only seems to happen on a dedicated server, it's fine on local saves. There are two gifs below that show the issues:
![Recording 2024-11-09 201306 (1)](https://github.com/user-attachments/assets/587d02ee-99b5-43be-a46e-9d194305ae8e)
![Recording 2024-11-09 201406 (1)](https://github.com/user-attachments/assets/03ff1032-d347-461c-ac6b-7d09cbd8ab09)

Tested with only Dank Storage and no other mods installed:
```
[20:30:40] [main/INFO]: Loading Minecraft 1.20.1 with Fabric Loader 0.16.9
[20:30:40] [main/INFO]: Dependencies overridden for interiors
[20:30:40] [main/INFO]: Loading 45 mods:
        - dankstorage 13
        - fabric-api 0.92.2+1.20.1
           |-- fabric-api-base 0.4.31+1802ada577
           |-- fabric-api-lookup-api-v1 1.6.36+1802ada577
           |-- fabric-biome-api-v1 13.0.13+1802ada577
           |-- fabric-block-api-v1 1.0.11+1802ada577
           |-- fabric-block-view-api-v2 1.0.1+1802ada577
           |-- fabric-blockrenderlayer-v1 1.1.41+1802ada577
           |-- fabric-client-tags-api-v1 1.1.2+1802ada577
           |-- fabric-command-api-v1 1.2.34+f71b366f77
           |-- fabric-command-api-v2 2.2.13+1802ada577
           |-- fabric-commands-v0 0.2.51+df3654b377
           |-- fabric-containers-v0 0.1.64+df3654b377
           |-- fabric-content-registries-v0 4.0.11+1802ada577
           |-- fabric-convention-tags-v1 1.5.5+1802ada577
           |-- fabric-crash-report-info-v1 0.2.19+1802ada577
           |-- fabric-data-attachment-api-v1 1.0.0+de0fd6d177
           |-- fabric-data-generation-api-v1 12.3.4+1802ada577
           |-- fabric-dimensions-v1 2.1.54+1802ada577
           |-- fabric-entity-events-v1 1.6.0+1c78457f77
           |-- fabric-events-interaction-v0 0.6.2+1802ada577
           |-- fabric-events-lifecycle-v0 0.2.63+df3654b377
           |-- fabric-game-rule-api-v1 1.0.40+1802ada577
           |-- fabric-item-api-v1 2.1.28+1802ada577
           |-- fabric-item-group-api-v1 4.0.12+1802ada577
           |-- fabric-key-binding-api-v1 1.0.37+1802ada577
           |-- fabric-keybindings-v0 0.2.35+df3654b377
           |-- fabric-lifecycle-events-v1 2.2.22+1802ada577
           |-- fabric-loot-api-v2 1.2.1+1802ada577
           |-- fabric-loot-tables-v1 1.1.45+9e7660c677
           |-- fabric-message-api-v1 5.1.9+1802ada577
           |-- fabric-mining-level-api-v1 2.1.50+1802ada577
           |-- fabric-model-loading-api-v1 1.0.3+1802ada577
           |-- fabric-models-v0 0.4.2+9386d8a777
           |-- fabric-networking-api-v1 1.3.11+1802ada577
           |-- fabric-networking-v0 0.3.51+df3654b377
           |-- fabric-object-builder-api-v1 11.1.3+1802ada577
           |-- fabric-particles-v1 1.1.2+1802ada577
           |-- fabric-recipe-api-v1 1.0.21+1802ada577
           |-- fabric-registry-sync-v0 2.3.3+1802ada577
           |-- fabric-renderer-api-v1 3.2.1+1802ada577
           |-- fabric-renderer-indigo 1.5.2+85287f9f77
           |-- fabric-renderer-registries-v1 3.2.46+df3654b377
           |-- fabric-rendering-data-attachment-v1 0.3.37+92a0d36777
           |-- fabric-rendering-fluids-v1 3.0.28+1802ada577
           |-- fabric-rendering-v0 1.1.49+df3654b377
           |-- fabric-rendering-v1 3.0.8+1802ada577
           |-- fabric-resource-conditions-api-v1 2.3.8+1802ada577
           |-- fabric-resource-loader-v0 0.11.10+1802ada577
           |-- fabric-screen-api-v1 2.0.8+1802ada577
           |-- fabric-screen-handler-api-v1 1.3.30+1802ada577
           |-- fabric-sound-api-v1 1.0.13+1802ada577
           |-- fabric-transfer-api-v1 3.3.5+8dd72ea377
           \-- fabric-transitive-access-wideners-v1 4.3.1+1802ada577
        - fabricloader 0.16.9
           \-- mixinextras 0.4.1
        - java 17
        - minecraft 1.20.1
```
Also, just noticed that it says "dankstorage 13" in the list above, but I can confirm I'm using "dankstorage-fabric-1.20.1-14.jar". 
~
Update: I built a new copy form source and this one does show up as "dankstorage 14", however still has both issues.

**Comment by Tfarcenim** [comment]
Fixed in 1.20.1-15, taking oversized stacks out of the dank is never intended.

**Comment by EnigmaticWraith** [comment]
It's been a bit, but I finally was able to test this out. I've installed the new 1.20.1-15 and the behavior seems to be almost unchanged, unfortunately. The only difference I'm seeing is I can't intentionally throw an oversized stack down by clicking outside of the window.

However, they still turn invisible on the cursor and they are still able to be accidentally dropped if an invisible oversize stack is on your cursor when you close the inventory or if you hit ctrl+q when hovering over it.
